servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/LoadBalancerObserver.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright © 2023 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.loadbalancer;
+
+import io.servicetalk.client.api.NoActiveHostException;
+import io.servicetalk.client.api.ServiceDiscovererEvent;
+
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+/**
+ * An observer that provides visibility into a {@link io.servicetalk.client.api.LoadBalancer}
+ * @param <ResolvedAddress> the type of the resolved address.
+ */
+interface LoadBalancerObserver<ResolvedAddress> {
```

Comment:
The current Observers all follow a different naming convention with an `on` prefix style name. Consider doing something similar here?

Reply:
Good point. This part of the change was from https://github.com/apple/servicetalk/pull/2770. I'll put up a separate PR to change the method names.

Reply:
PR: https://github.com/apple/servicetalk/pull/2783